num/quat/quat.go

```diff
@@ -0,0 +1,105 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package quat
+
+import (
+	"bytes"
+	"fmt"
+)
+
+var zero Quat
+
+// Quat is a float64 precision quaternion.
+type Quat struct {
+	Real, Imag, Jmag, Kmag float64
```

Comment:
Another possible set of field names are R, I, J and K.

Reply:
During the review I got used to the `mag` version, initially I thought that `Jmag` and `Kmag` carry too much cleverness and play-on-wordiness.

Reply:
Yes, recall that the original code this is based on was an April Fool's joke.

I'd change the suggestion above so that it's Real instead of R.

However, there is an alternative pronunciation that makes it sensible this way. Most people read `imag` is `imag(inary)`, but it could easily be `imag(nitude)`. This second pronunciation would probably be clearer that I, J and K.

Reply:
I also agree with the `mag` version although the connection to magnitude is unfortunate because magnitude is inherently positive. But since Go has already `real` and `imag`, and since quaternions are an extension of complex numbers, it makes sense to continue the trend and have `Jmag` and `Kmag`. For me other convincing arguments are that `I, J, K` are small enough to get lost in source and that they are not the quaternion units as the name would suggest.